cbcontainers/state/state_applier.go

```diff
@@ -146,22 +148,20 @@ func (c *StateApplier) applyEnforcer(ctx context.Context, agentSpec *cbcontainer
 
 	mutatedService, _, err := c.applier.Apply(ctx, c.enforcerService, agentSpec, applyOptions)
 	if err != nil {
-		if deleted, deleteErr := c.applier.Delete(ctx, c.enforcerWebhook, agentSpec); deleteErr != nil {
+		if _, deleteErr := c.deleteAllEnforcerWebhooks(ctx, agentSpec); deleteErr != nil {
 			return false, deleteErr
-		} else if deleted {
-			c.log.Info("Deleted enforcer webhook")
 		}
+
 		return false, err
 	}
 	c.log.Info("Applied enforcer service", "Mutated", mutatedService)
 
 	mutatedDeployment, deploymentK8sObject, err := c.applier.Apply(ctx, c.enforcerDeployment, agentSpec, applyOptions)
 	if err != nil {
-		if deleted, deleteErr := c.applier.Delete(ctx, c.enforcerWebhook, agentSpec); deleteErr != nil {
+		if _, deleteErr := c.deleteAllEnforcerWebhooks(ctx, agentSpec); deleteErr != nil {
```

Comment:
we can't ignore this, we need to another change, we need to separate it from the if condition, and send the deleted variable as part of the function result, so we return true to the controller to show that the state was changed (component deleted)

Reply:
Ok, we can change that - but this is not new behavior
Observe that in the old code, `deleted` is only used to log a message. It does not propagate up the stack as well - so maybe we had that issue even before this change?